src/orchestrator/orchestrator.rs

```diff
@@ -103,58 +131,76 @@ impl<const N_ASSETS: usize, const N_BYTES: usize> Orchestrator<N_ASSETS, N_BYTES
             // - Parses CSV file into 'entries'.
             // - Sends 'entries' to executors via [entries_tx] channel.
             //
-            let (start, end) = self.calculate_task_range(i, number_executor);
+            let (start, end) = self.calculate_task_range(i, executor_count);
             let entry_csvs_slice = self.entry_csvs[start..end].to_vec(); // Clone only the necessary slice
 
+            // let stop_tx_clone = stop_tx.clone();
+            let cloned_cancel_token = cancel_token.clone();
             tokio::spawn(async move {
                 for file_path in entry_csvs_slice.iter() {
-                    let entries = entry_parser::<_, N_ASSETS, N_BYTES>(file_path).unwrap();
-                    if entries_tx.send(entries).await.is_err() {
-                        break;
+                    let entries = match entry_parser::<_, N_ASSETS, N_BYTES>(file_path) {
+                        Ok(entries) => entries,
+                        Err(e) => {
+                            eprintln!("Executor_{:?}: Error while processing file {:?}: {:?}", i, file_path, e);
+                            // let _ = stop_tx_clone.send(()).unwrap();
+                            cloned_cancel_token.cancel();
+                            break;
+                        }
+                    };
+
+                    tokio::select! {
+                        _ = cloned_cancel_token.cancelled() => {
+                            eprintln!("Executor_{:?}: cancel signal received, terminating distributor.", i);
+                            break;
+                        },
+                        send_entries = entries_tx.send(entries) => {
```

Comment:
Can we be sure here that the sending completes before the cancellation signal is received? Since there's a select, I assume that there can be a situation when the cancellation token is received before the sending is finished.

Reply:
Your assumption is correct. If the cancellation token is received during the select process, it will indeed cause an immediate cancellation, even while a task is being processed. 
 However, Discarding the entire process(generating mini tree) when a cancellation token is received simplifies the system, as adding recovery mechanisms for failed points would significantly increase complexity.
 I plan to give this more thought and consider potential improvements in the future.
 
 Fixed cancel points at [089c744](https://github.com/summa-dev/summa-aggregation/pull/3/commits/089c744ced3bc2f3a8e16827c026abb73574bf0f)